code/modules/mob/living/simple_animal/hostile/terror_spiders/actions.dm

```diff
@@ -0,0 +1,127 @@
+	var/mob/living/simple_animal/hostile/poison/terror_spider/user = owner
+	user.Web()
+
+/datum/action/innate/terrorspider/wrap
+	name = "Wrap"
+	icon_icon = 'icons/effects/effects.dmi'
+	button_icon_state = "cocoon_large1"
+
+/datum/action/innate/terrorspider/wrap/Activate()
+	var/mob/living/simple_animal/hostile/poison/terror_spider/user = owner
+	user.FindWrapTarget()
+	user.DoWrap()
+
+
+// ---------- WEB
+
+/mob/living/simple_animal/hostile/poison/terror_spider/proc/Web()
+	visible_message("<span class='notice'>\the [src] begins to secrete a sticky substance.</span>")
+	if(do_after(src, 40, target = loc))
+		new /obj/effect/spider/terrorweb(loc)
+
+/obj/effect/spider/terrorweb
+	name = "terror web"
+	desc = "it's stringy and sticky"
+	icon = 'icons/effects/effects.dmi'
+	anchored = 1 // prevents people dragging it
+	density = 0 // prevents it blocking all movement
+	health = 20 // two welders, or one laser shot (15 for the normal spider webs)
+	icon_state = "stickyweb1"
+
+
+/obj/effect/spider/terrorweb/New()
+	if(prob(50))
+		icon_state = "stickyweb2"
+
+
+/obj/effect/spider/terrorweb/CanPass(atom/movable/mover, turf/target, height=0, air_group=0)
+	if(air_group || (height==0)) return 1
+	if(istype(mover, /mob/living/simple_animal/hostile/poison/terror_spider))
+		return 1
+	if(istype(mover, /mob/living))
+		if(prob(80))
+			to_chat(mover, "<span class='danger'>You get stuck in \the [src] for a moment.</span>")
+			var/mob/living/M = mover
+			M.Stun(4) // 8 seconds.
+			M.Weaken(4) // 8 seconds.
+			return 1
+		else
+			return 0
+	if(istype(mover, /obj/item/projectile))
+		return prob(20)
+	return ..()
+
+
+
+
+// ---------- WRAP
+
+
+/mob/living/simple_animal/hostile/poison/terror_spider/proc/FindWrapTarget()
+	if(!cocoon_target)
+		var/list/choices = list()
+		for(var/mob/living/L in oview(1,src))
+			if(Adjacent(L))
+				if(L.stat == DEAD)
+					choices += L
+		for(var/obj/O in oview(1,src))
+			if(Adjacent(O) && !O.anchored)
+				if(!istype(O, /obj/effect/spider/terrorweb) && !istype(O, /obj/effect/spider/cocoon))
+					choices += O
+		if(choices.len)
+			cocoon_target = input(src,"What do you wish to cocoon?") in null|choices
+		else
+			to_chat(src, "<span class='danger'>There is nothing nearby you can wrap.</span>")
+
+/mob/living/simple_animal/hostile/poison/terror_spider/proc/DoWrap()
+	if(cocoon_target && busy != SPINNING_COCOON)
+		busy = SPINNING_COCOON
+		visible_message("<span class='notice'>\the [src] begins to secrete a sticky substance around \the [cocoon_target].</span>")
+		stop_automated_movement = 1
+		walk(src,0)
+		if(do_after(src, 40, target = cocoon_target.loc))
+			if(busy == SPINNING_COCOON)
+				if(cocoon_target && istype(cocoon_target.loc, /turf) && get_dist(src,cocoon_target) <= 1)
+					var/obj/effect/spider/cocoon/C = new(cocoon_target.loc)
+					var/large_cocoon = 0
+					C.pixel_x = cocoon_target.pixel_x
+					C.pixel_y = cocoon_target.pixel_y
+					for(var/obj/item/I in C.loc)
```

Comment:
I just did it this way because I saw other code elsewhere that did. The obvious alternative for the three obj types is simply looping over /obj/ and then doing istype(/obj/item) || istype(/obj/machinery/) || istype(/obj/structure). Would that be better? If there's a specific way you want it, please say.